main.go

```diff
@@ -333,11 +335,35 @@ func checkEmptyArgs(args ...string) bool {
 }
 
 // choose http or https
-func serve(logger log.Logger, tls bool, port, key, cert string) {
+func serve(logger log.Logger, tlsEnabled bool, port, key, certPath string) {
 	portStr := fmt.Sprintf(":%v", port)
-	if tls {
+	if tlsEnabled {
+		chain, err := tls.LoadX509KeyPair(certPath, key)
+		if err != nil {
+			logger.Log("err", "failed to load TLS certificate or private key")
+			os.Exit(1)
+		}
+
+		cert, err := x509.ParseCertificate(chain.Certificate[0]) // Leaf is always the first entry
+		if err != nil {
+			logger.Log("err", "error parsing TLS certificate")
+			os.Exit(1)
+		}
+
+		if _, err := cert.Verify(x509.VerifyOptions{}); err != nil {
```

Comment:
I'm not sure but this is lifted from buford verbatim :)
I don't know what extra roots i could provide other than our included CA?